src/StringHelper.php

```diff
@@ -577,4 +579,87 @@ static function (string $key) use ($delimiter, $escapeCharacter): string {
             $result
         );
     }
+
+    /**
+     * Strip Unicode whitespace (with property White_Space=yes) or other characters from the beginning and end of a string.
+     * Input string and pattern are treated as UTF-8.
+     *
+     * @see https://en.wikipedia.org/wiki/Whitespace_character#Unicode
+     * @see https://www.php.net/manual/function.preg-replace
+     *
+     * @param string|string[] $string The string or an array with strings.
+     * @param string $pattern PCRE regex pattern to search for, as UTF-8 string. Quote $pattern if it contains special regular expression characters.
+     * @see https://www.php.net/manual/function.preg-quote.php
+     *
+     * @psalm-template TKey of array-key
+     * @psalm-param string|array<TKey, string> $string
+     * @psalm-param non-empty-string $pattern
+     * @psalm-return ($string is array ? array<TKey, string> : string)
+     *
+     * @return string|string[]
+     */
+    public static function trim(string|array $string, string $pattern = self::DEFAULT_WHITESPACE_PATTERN): string|array
```

Comment:
```suggestion
    public static function trim(string ...$string, string $pattern = self::DEFAULT_WHITESPACE_PATTERN): string|array
```
WDYT?

Reply:
Can you show the case with `trim()` call like `trim($str1, $str2)`?
I think transformation functions like `trim()` must keep source data structure, because they are usually used for data preprocessing rather than computation.
```
$rawData = ['param1' => ' dd', 'param2' => 'ff '];
$data = trim($rawData);
```